public/app/features/logs/components/LogRowMessageDisplayedFields.tsx

```diff
@@ -25,32 +25,28 @@ export interface Props {
 
 export const LogRowMessageDisplayedFields = React.memo((props: Props) => {
   const { row, detectedFields, getFieldLinks, wrapLogMessage, styles, mouseIsOver, pinned, ...rest } = props;
-  const fields = getAllFields(row, getFieldLinks);
   const wrapClassName = wrapLogMessage ? '' : displayedFieldsStyles.noWrap;
+  const fields = useMemo(() => getAllFields(row, getFieldLinks), [getFieldLinks, row]);
   // only single key/value rows are filterable, so we only need the first field key for filtering
-  const line = useMemo(
-    () =>
-      detectedFields
-        .map((parsedKey) => {
-          const field = fields.find((field) => {
-            const { keys } = field;
-            return keys[0] === parsedKey;
-          });
+  const line = useMemo(() => {
+    let line = '';
+    for (let i = 0; i < detectedFields.length; i++) {
+      const parsedKey = detectedFields[i];
+      const field = fields.find((field) => {
```

Comment:
Anything that can speed this up would be amazing. I'll see if that makes a difference, but the reason why this is super slow is because of `getAllFields()`. It does a lot of things inside, even parsing the `DataFrame` to a `LogsFrame`.